### PR TITLE
Add systemd timer support

### LINKED ISSUE #48: Add support to systemd timers
References to timers:
  - http://0pointer.de/public/systemd-man/systemd.timer.html
  - https://www.freedesktop.org/software/systemd/man/systemd.timer.html

Examples:
  - https://wiki.archlinux.org/index.php/Systemd/Timers#Examples

### PR DESCRIPTION
**Pull request by komish**
### Provides Systemd Timer and associated service file for optional use.

Fixes #48 

### Considerations
* As far as frequency, `OnCalendar=daily` is being used to match the provided cron.daily file. This was tested using `OnCalendar=minutely` or equivalent (detailed below).
* The `.service` and `.timer` files have been tested with `systemd-analyze` where possible. This may be worth including in tests at a later time - if the test suite of choice has the capability.
* Tested against the following (python2.x provided by distribution repositories, kthresher installed via setup.py from source/development branch):
    * Ubuntu 16.04 
        * Versions
            * Python 2.7.12
            * Systemd 229-4ubuntu21.4
    * Ubuntu 18.04 
        * Versions
            * Python 2.7.15rc1
            * Systemd 237-3ubuntu10.3
    * Debian Jessie 
        * Version:
            * Python 2.7.9
            * Systemd 215-17+deb8u7
        * Notes
            * systemd does not allow for `OnCalendar=minutely` in this version, however daily is an option so `*-*-* *:*:00` was used for testing to simulate the same behavior as `minutely`.
            * `systemd-analyze` did not have a `verify` option in this version.
    * Debian Stretch 
        * Versions
            * Python 2.7.13
            * Systemd 232-25+deb9u4
* In testing, installation via setup.py results in install path at `/usr/local/bin/kthresher` - I've left `/usr/bin/kthresher` as the path to use to match the cron.daily example but testing was done at the specified path. Shouldn't impact anything, but we cannot dynamically build the path given the current installation procedure.
* From what I can see, the cron.daily entry should fail with a default configuration as there are no options specified (as in, help output should be displayed and a non-zero exit code should return). The default systemd service definition will have the --dry-run option such that by default nothing will happen, but the timer and service will not fail. This is mostly a precaution.

### CODE REVIEW

File: kthresher.systemd.service [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[Unit]
+Description=Purge unused kernels
+Documentation=man:kthresher(8)
+
+[Service]
+Type=oneshot
+# by default, do nothing but provide information
+ExecStart=/usr/bin/kthresher --dry-run
```
Comment (tonyskapunk) [inline kthresher.systemd.service]: It's OK to *not* use `--dry-run` the [default config](https://github.com/rackerlabs/kthresher/blob/master/kthresher.conf) does not purge any package unless it is defined.

**Review by tonyskapunk — CHANGES_REQUESTED** [review]
Thanks for providing these, let's remove the `--dry-run` as that will be handled by the config.

**Comment by komish** [comment]
@tonyskapunk 

The issue is that the default config does not specify an action to take from what I see, so the service reports as failed because the command with no arguments/config returns a non-zero exit code

```
# systemctl status kthresher.service
● kthresher.service - Purge unused kernels
   Loaded: loaded (/etc/systemd/system/kthresher.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sun 2018-10-07 23:58:01 UTC; 25s ago
     Docs: man:kthresher(8)
  Process: 4759 ExecStart=/usr/local/bin/kthresher (code=exited, status=1/FAILURE)
 Main PID: 4759 (code=exited, status=1/FAILURE)

Oct 07 23:58:01 ubuntu systemd[1]: kthresher.service: Main process exited, code=exited, status=1/FAILURE
Oct 07 23:58:01 ubuntu systemd[1]: kthresher.service: Failed with result 'exit-code'.
Oct 07 23:58:01 ubuntu systemd[1]: Failed to start Purge unused kernels.
```

```
# kthresher &>/dev/null; echo $?
1
```

If that's fine, I'll remove the `--dry-run`. Alternatives might include leaving `--dry-run`, or specifying the `ExecStart` line have a `-` before the executable here. This causes systemd to ignore the non-zero and report success (or in this case, inactivity). I'm not sure that's a better path - when we get non-zero, I would imagine we want this to report failure.

```
ExecStart=-/usr/local/bin/kthresher
```

```
# systemctl status kthresher.service
● kthresher.service - Purge unused kernels
   Loaded: loaded (/etc/systemd/system/kthresher.service; static; vendor preset: enabled)
   Active: inactive (dead) since Mon 2018-10-08 00:00:00 UTC; 8s ago
     Docs: man:kthresher(8)
  Process: 4874 ExecStart=/usr/local/bin/kthresher (code=exited, status=1/FAILURE)
 Main PID: 4874 (code=exited, status=1/FAILURE)
 ```

 Let me know which you would prefer.

**Comment by komish** [comment]
The config I'm seeing as "default" for reference

```# cat /etc/kthresher.conf
[main]
include = /etc/kthresher.d/*.conf

# Is recommended to put the configuration in the included directory instead
# of making changes in here.
#
# The options are:
#   headers - [ yes | on | true | no | off | false ] Include or not the headers.
#   include - [ /path/to/file ] Suports globbing, e.g. /path/to/dir/*.conf
#   keep - Number of kernel images to keep (0-9), default is 1 if none defined
#   purge - [ yes | on | true | no | off | false ] Commit changes
#   verbose - [ yes | on | true | no | off | false ] Be verbose
#
# NOTES:
# 1) The --dry-run option is only allowed through CLI, if purge is defined
#    in the config file and --dru-run is used, the latter has precedence.
# 2) IF defined, only the non-boolean options can be overriden from CLI.```

There is nothing in the related `kthresher.d` out of the box.

**Comment by tonyskapunk** [comment]
For now let's remove the `--dry-run` I'll make the changes to avoid a failure. Thanks @Komish

**Comment by komish** [comment]
@tonyskapunk Done - I've removed the comment and the `--dry-run` flag. Should be set to go. Let me know if there is anything further!

**Comment by tonyskapunk** [comment]
Thaaaanks!, there will be stickers! :+1:

**Pull request merged**